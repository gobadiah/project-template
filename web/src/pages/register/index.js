import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
