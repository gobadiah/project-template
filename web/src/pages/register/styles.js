import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
