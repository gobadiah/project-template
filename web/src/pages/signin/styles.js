import { css } from 'react-emotion';

// eslint-disable-next-line import/prefer-default-export
export const containerClassName = css`
  width: 640px;
  height: 682px;
`;
