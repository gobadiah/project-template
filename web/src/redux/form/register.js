import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
