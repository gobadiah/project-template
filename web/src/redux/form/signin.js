import React from 'react';
import { arrayOf, func, string } from 'prop-types';

import { reduxForm } from 'redux-form';

import { defaultPropTypes } from '~/components';

import {
  formatError as defaultFormatError,
  formatWarning as defaultFormatWarning,
} from './utils';

import { EmailField, PasswordField } from './fields';

const SignInForm = ({
  error,
  formatError,
  formatWarning,
  handleSubmit,
  warning,
}, { t }) => (
  <form onSubmit={handleSubmit}>
    <EmailField />
    <PasswordField />
    <input type='submit' value={t('Sign in')} />
    {formatWarning(warning, t)}
    {formatError(error, t)}
  </form>
);

SignInForm.propTypes = {
  error: arrayOf(string),
  formatError: func,
  formatWarning: func,
  handleSubmit: func.isRequired,
  warning: arrayOf(string),
};

SignInForm.defaultProps = {
  error: [],
  formatError: defaultFormatError,
  formatWarning: defaultFormatWarning,
  warning: [],
};

SignInForm.contextTypes = defaultPropTypes;

export { SignInForm as PlainSignInForm };

export default reduxForm({
  form: 'SignInForm',
})(SignInForm);
