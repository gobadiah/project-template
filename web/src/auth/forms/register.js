import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
