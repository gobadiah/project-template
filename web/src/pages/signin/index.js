import React from 'react';
import { ToastContainer } from 'react-toastify';

import { FacebookButton, GoogleButton, Main } from '~/components';
import { Line, SocialDiv } from '~/styles';
import { Page } from '~/components/base';
import { SignInForm } from '~/auth';
import hoc from '~/hoc';
import { Link } from '~/routes';

import { containerClassName } from './styles';

class SignIn extends Page {
  render() {
    const { t } = this.props;
    const { signin } = this.props;
    return (
      <Main
        title={t('sign-in:title')}
        titleBar={t('sign-in:Connect to your account')}
        containerClassName={containerClassName}
      >
        <SocialDiv>
          <GoogleButton />
          <FacebookButton />
        </SocialDiv>
        <Line />
        <SignInForm onSubmit={signin} />
        <Line />
        <Link route='register'>
          <a>{t('sign-in:Create your account')}</a>
        </Link>
      </Main>
    );
  }
}

export default hoc('sign-in', { needsLogin: false })(SignIn);
