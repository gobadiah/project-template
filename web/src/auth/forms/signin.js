import React from 'react';
import { arrayOf, func, string } from 'prop-types';

import { reduxForm } from 'redux-form';

import { defaultPropTypes } from '~/components';

import { Error, Warning } from '~/components/forms';

import { EmailField, PasswordField } from '~/components/forms/fields';

const SignInForm = ({
  error,
  handleSubmit,
  warning,
}, { t }) => (
  <form onSubmit={handleSubmit}>
    <EmailField />
    <PasswordField />
    <input type='submit' value={t('Sign in')} />
    <Warning warning={warning} />
    <Error error={error} />
  </form>
);

SignInForm.propTypes = {
  error: arrayOf(string),
  handleSubmit: func.isRequired,
  warning: arrayOf(string),
};

SignInForm.defaultProps = {
  error: [],
  warning: [],
};

SignInForm.contextTypes = defaultPropTypes;

export { SignInForm as PlainSignInForm };

export default reduxForm({
  form: 'SignInForm',
})(SignInForm);
