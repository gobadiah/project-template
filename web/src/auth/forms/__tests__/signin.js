import 'jsdom-global/register';

import React from 'react';
import { shallow } from 'enzyme';

import { PlainSignInForm } from '../signin';

const store = {
  getState: jest.fn(() => ({
    form: {},
  })),
  subscribe: jest.fn(),
  dispatch: jest.fn(),
};

const context = {
  t: jest.fn(),
  i18n: {},
  store,
};

describe('PlainSignInForm', () => {
  it('should match snapshot', () => {
    const handleSubmit = jest.fn();
    const tree = shallow(<PlainSignInForm handleSubmit={handleSubmit} />, { context });
    expect(tree).toMatchSnapshot();
  });
});
