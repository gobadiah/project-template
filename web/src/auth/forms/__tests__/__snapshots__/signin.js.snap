// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainSignInForm should match snapshot 1`] = `
.emotion-0 {
  width: 520px;
}

.emotion-0 > * {
  margin-bottom: 23px;
}

<Form
  className="emotion-0"
  error={Array []}
  handleSubmit={[MockFunction]}
  warning={Array []}
>
  <Component />
  <Component />
  <Styled(div)>
    <LinkRoutes
      route="password-forgotten"
    >
      <Styled(a) />
    </LinkRoutes>
    <RoundButton
      className=""
      text=""
      type="submit"
    />
  </Styled(div)>
</Form>
`;
